lib/assets/javascripts/_modules/search.js

```diff
@@ -149,7 +151,8 @@
         var content = s.processContent(result.content, query)
         output += '<li class="search-result">'
         output += '<h3 class="search-result__title">'
-        output += '<a href="' + result.url + '">'
+        var url = result.url.startsWith('/') ? result.url.slice(1) : result.url
+        output += '<a href="' + pathToSiteRoot + url + '">'
```

Comment:
Ah, sorry, my previous suggestion broke things slightly 😅 

It turns out that the search index in `search.json` includes `http_prefix` in the URLs, so when using http prefix and not relative links we now end up with the http prefix duplicated 🤔 .

Reply:
I think the easiest way to fix this is to check if `pathToSiteRoot` is a relative link, and only append if it is

```suggestion
        url = pathToSiteRoot.startsWith('..') ? pathToSiteRoot + url : '/' + url  // url already contains pathToSiteRoot when using :http_prefix
        output += '<a href="' + url + '">'
```

There might be a neater way to do that though 😅

Reply:
Aah, good spot @lfdebrux ! Will take a look. Out of interest, have you been able to get a site running locally when specifying `http_prefix`? All the sites I've tried break horribly as none of the assets are available under their prefixed URLs :thinking: .

Reply:
hmm, I'm a little confused by this. I _think_ it has been caused by this change in `path_helpers.rb`:

```ruby
path_to_site_root = config[:http_prefix]
```

After the change, this results in any configured `http_prefix` being added in the path to the site root, which is then fed down in to the search system, breaking the search.

[This (rather historical) comment](https://github.com/middleman/middleman/issues/484#issuecomment-6459631) seems to suggest that `http_prefix` is only intended for use to prefix assets (for example when using a CDN and needing to specify some prefix). Having read that I'm now feeling that our use of it to calculate the site root is contrary to its intended use. Unfortunately the issue that comment comes from was never resolved, despite nearly a year of discussion :disappointed:.

I think my preference would be to revert back to reporting the site root as `/` in `path_helpers.rb` when using absolute links, this feels more in keeping with the intent of `http_prefix` to me. Grateful for your thoughts @lfdebrux .

Reply:
> Out of interest, have you been able to get a site running locally when specifying http_prefix? All the sites I've tried break horribly as none of the assets are available under their prefixed URLs 🤔 .

Yes, the assets do cause issues, however we recently closed one PR to fix that partially #197, and I have a PR open to fix that for govuk-frontend assets #292. There's also a fix needed for the path `search.json`, in #196, although I'm hoping that we can fix that for both http_prefix and relative_links in this PR 🤞. 

If you're interested in how I've configured everything, right now I have a couple of copies of the example site building together, with different configs, and I serve them from one directory using a Python test server.

In short what I'm doing is:

```
cd example
bundle exec middleman build --http-prefix=/with-http-prefix/
cd ../..
mkdir public
cp -r tech-docs-gem/example/build public/with-http-prefix
cd public
python3 -m http.server
```

That seems to work for me, if I have the right fixes in place. I think copying (or symbolic linking) the build into a folder with the same name as the http_prefix is important; it might also be important to make sure the http_prefix begins and ends with a slash, although I haven't played around with that too much.

Reply:
Sorry for the delay in replying @lfdebrux , thanks for the tips on getting things running with an `http_prefix`, that was really helpful. I have just pushed what I hope is an appropriate fix to ensure that `http_prefix` is no longer duplicated when we generate the links to the search result pages. I have tested it in all the use cases I can think of (relative links, absolute links with and without an http_prefix) but would be grateful if you could give it some :eyes: too. Cheers!

Reply:
Awesome, thanks! I'll take a look 👀